cmd/ir2proxy/main.go

```diff
@@ -38,31 +38,36 @@ func main() {
 		log.Fatal(err)
 	}
 
-	contourscheme.AddToScheme(scheme.Scheme)
-
+	// Split the yaml file by docs, and parse each one individually
 	for _, yamldoc := range bytes.Split(data, []byte("---")) {
 		if len(yamldoc) == 0 {
 			continue
 		}
 
-		decode := scheme.Codecs.UniversalDeserializer().Decode
-
-		ir, groupVersionKind, err := decode(yamldoc, nil, nil)
+		ir, err := k8sdecoder.DecodeIngressRoute(yamldoc)
 		if err != nil {
-			log.Infof("Skipped yaml doc, %s", err)
+			log.Error(err)
 			continue
 		}
-		switch t := ir.(type) {
-		case *irv1beta1.IngressRoute:
-			log.Infof("IngressRoute %s, namespace %s, labels %s", t.ObjectMeta.Name, t.ObjectMeta.Namespace, t.ObjectMeta.Labels)
-			hp, err := translateIngressRouteToHTTPProxy(t)
-			if err != nil {
-				log.Fatal(err)
+
+		validateWarnings := validate.CheckIngressRoute(ir)
+		if len(validateWarnings) > 0 {
+			for _, warning := range validateWarnings {
+				log.Error(warning)
+			}
+		}
+		hp, warnings := translate.IngressRouteToHTTPProxy(ir)
+		if len(warnings) > 0 {
+			for _, warning := range warnings {
+				log.Error(warning)
 			}
-			log.Error(spew.Sdump(hp))
-		default:
-			log.Errorf("This utility only works with IngressRoute, a %s was supplied.", groupVersionKind)
 		}
+		outputYAML, err := yaml.Marshal(hp)
+		if err != nil {
+			log.Error(err)
```

Comment:
given that we control the output being serialised it seems unlikely that this error will be frequent. The program should probably stop at this point.